dsgrid/registry/project_registry_manager.py

```diff
@@ -292,34 +294,57 @@ def _check_dataset_base_to_project_base_mappings(
         types = [x for x in DimensionType if x not in exclude_dims]
         dimension_pairs = [tuple(sorted((x, y))) for x, y in itertools.combinations(types, 2)]
 
-        data_source = dataset_config.model.data_source
+        data_source_dim_id = [
+            x.id for x in dataset_config.dimensions if x.type == DimensionType.DATA_SOURCE
+        ][0]
+        data_source = [
+            x.id for x in self._dimension_mgr.get_by_id(data_source_dim_id).model.records
+        ][
+            0
+        ]  # this assumes only only data_source per dataset
         dim_table = handler.get_unique_dimension_rows().drop("id")
+        for col in dim_table.columns:
+            assert dim_table.filter(F.col(col).isNull()).count() == 0
```

Comment:
Should this be an exception instead of assert? Seems like we could use this as a helper function.